extensions/wikia/Blogs/BlogArticle.php

```diff
@@ -135,66 +145,101 @@ private function showBlogListing() {
 	 * @access public
 	 */
 	public function clearBlogListing() {
-		global $wgRequest, $wgMemc, $wgLang;
+		global $wgMemc;
 
 		// Clear Oasis rail module
-		$mcKey = wfMemcKey( "OasisPopularBlogPosts", $wgLang->getCode() );
+		$mcKey = $this->blogListingOasisMemcacheKey();
 		$wgMemc->delete( $mcKey );
 
-		$user = $this->mTitle->getPrefixedDBkey();
-		foreach ( range( 0, 5 ) as $page ) {
-			$wgMemc->delete( $this->blogListingMemcacheKey( $user, $page ) );
+		$count = $this->getBlogListingPageCount();
+		foreach ( range( 0, $count - 1 ) as $page ) {
+			$mcKey = $this->blogListingMemcacheKey( $this->getBlogOwner(), $page );
+			$wgMemc->delete( $mcKey );
 		}
+
+		/** @var BlogArticle|WikiPage $this */
 		$this->doPurge();
 
 		$title = Title::newFromText( 'Category:BlogListingPage' );
 		$title->touchLinks();
+	}
+
```

Comment:
The method below uses the parser to parse a 'bloglist' tag specifying that a count of blog post to be returned.  Generating this count is tightly integrated with parsing this tag and I couldn't find a good way to call this directly without a total overhaul of the code.